From:	Simon Wunderlich <simon.wunderlich@s2003.tu-chemnitz.de>
Subject: [PATCH] mac80211: adapt slot time in IBSS mode
Date:	Mon,  3 Dec 2012 22:21:30 +0100
Message-Id: <1354569690-23559-1-git-send-email-siwu@hrz.tu-chemnitz.de>

In 5GHz/802.11a, we are allowed to use short slot times. Doing this
may increases performance by 20% for legacy connections (54 MBit/s).
I can confirm this in my tests (27% more throughput using iperf), and
also have a small positive effect (5% more throughput) for HT rates,
tested on 1 stream.

Signed-off-by: Simon Wunderlich <siwu@hrz.tu-chemnitz.de>
Signed-off-by: Mathias Kretschmer <mathias.kretschmer@fokus.fraunhofer.de>
[sven@open-mesh.com: Enable it also on 2.4 Ghz]
Signed-off-by: Sven Eckelmann <sven@open-mesh.com>

Forwarded: not-needed
 This is only allowed for 5G but used as by Open Mesh for 2.4G to increase
 throughput.
---
 net/mac80211/ibss.c |   13 +++++++++++++
 1 file changed, 13 insertions(+)

--- a/net/mac80211/ibss.c
+++ b/net/mac80211/ibss.c
@@ -213,7 +213,7 @@ static void __ieee80211_sta_join_ibss(st
 	 *
 	 * HT follows these specifications (IEEE 802.11-2012 20.3.18)
 	 */
-	sdata->vif.bss_conf.use_short_slot = chan->band == IEEE80211_BAND_5GHZ;
+	sdata->vif.bss_conf.use_short_slot = 1;
 	bss_change |= BSS_CHANGED_ERP_SLOT;
 
 	sdata->vif.bss_conf.ibss_joined = true;
